### PR TITLE
Adds Prisoner Network system with Azure AD for use by Prisoner Content Hub

### PR DESCRIPTION
**Pull request by spikeheap**
## What does this pull request do?

This adds the Azure Active Directory service used for authenticating in-cell devices and the Prisoner Content Hub:

structurizr-56937-prisonerContentHubContainer

## What is the intent behind these changes?

To model more deployed services as code 🥳 .

### CODE REVIEW

File: src/main/kotlin/model/PrisonerNetwork.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class PrisonerNetwork private constructor() {
+  companion object : HMPPSSoftwareSystem {
+    lateinit var system: SoftwareSystem
+    lateinit var auth: Container
+
+    override fun defineModelEntities(model: Model) {
+      system = model.addSoftwareSystem("Prisoner Network", "Provides in-cell devices and access to digital services to prisoners")
+
+      auth = system.addContainer(
+        "Azure AD",
```
Comment (sldblog) [inline src/main/kotlin/model/PrisonerNetwork.kt:19]: I'm wondering, is this container shared in HMPPS for other purposes, or specifically provisioned for this use case?

Comment (spikeheap) [reply src/main/kotlin/model/PrisonerNetwork.kt]: Great question. This is a component of the Digital Prisoner Network, and AD was provisioned to manage prisoner device login (the in-cell devices are part of this system too).

We've just started using AD as an Oauth2 provider to authenticate prisoners on the content hub. I'm sure we'll see more use for this in future across HMPPS (and why not the wider MoJ?)

Comment (sldblog) [reply src/main/kotlin/model/PrisonerNetwork.kt]: Thanks! For the wider MoJ, there's a "DOM1" Active Directory for internal staff -- I was wondering whether this is the same?

Comment (sldblog) [reply src/main/kotlin/model/PrisonerNetwork.kt]: What do you think about naming this as "Prisoner authentication service"?

Comment (spikeheap) [reply src/main/kotlin/model/PrisonerNetwork.kt]: I'd like to keep naming as close to the terminology that team is using as a starting point, but I'm not too tied to that approach. How would you feel about us updating this as it matures from a system component into a shared service? I've got some upcoming conversations with the Access team which will help me understand if/how this fits with their plans which might prompt some changes.

Comment (mxco86) [reply src/main/kotlin/model/PrisonerNetwork.kt]: It looks like naming as 'Prisoner Authentication Service'is a better fit with the rest of the component names. If we could add the extra info to distinguish it from the DOM1 AD in the description that would also help. When we add the authentication systems with the Access work we can use the same form.

**Review by sldblog — APPROVED** [review]

**Pull request merged**